R/class2tree.R

```diff
@@ -310,27 +315,46 @@ rank_indexing <- function (rankList) {
         # for old rank
         if (i > 1) {
           if (flag == 0) {
+            if (!(iRank %in% ls(rank2index))) stop(paste(iRank,"not found!"))
             currentIndex <- rank2index[[iRank]]
           } else {
+            if (!(iRank %in% ls(tmpEnv))) {
+              tmpEnv[[iRank]] <- tmpEnv[[subList[i-1]]]
+            }
             currentIndex <- tmpEnv[[iRank]]
           }
+          
           if (currentIndex <= tmpEnv[[subList[i-1]]]) {
             if (flag == 0) {
               tmpEnv[[iRank]] <- tmpEnv[[subList[i-1]]] + 1
-              for (
-                r in ls(rank2index)[!(ls(rank2index) %in% ls(tmpEnv))]
-              ) {
-                if (rank2index[[r]] > currentIndex) {
-                  tmpEnv[[r]] <- 
+              # list of current ranks that whose index should be increased
+              candidateList <- unlist(
+                mget(
+                  ls(rank2index)[!(ls(rank2index) %in% ls(tmpEnv))], rank2index
+                )
+              )
+              candidateList <- candidateList[order(unlist(candidateList))]
+              for (c in seq_len(length(candidateList))) {
```

Comment:
maybe do not use `c` as a for loop index since its already a function in base R?

Reply:
done!